src/main/resources/bamm/meta-model/2.0.0/aspect-meta-model-shapes.ttl

```diff
@@ -201,32 +216,43 @@ bamm:EntityShape
                 group by $this
               }
               filter ( !bound( ?propertyCount ) || ( ?propertyCount = 0 ) )
+              bind( 'ERR_INVALID_PROPERTY' as ?code )
             }
          """
       ] ;
       sh:sparql [
          a sh:SPARQLConstraint ;
-         sh:message "A Property of an Entity may not be defined with both bamm:optional \"true\"^^xsd:boolean and bamm:notInPayload \"true\"^^xsd:boolean." ;
+         sh:message "Property '{?value}' of the Entity '{$this}' may not be defined with both bamm:optional \"true\"^^xsd:boolean and bamm:notInPayload \"true\"^^xsd:boolean." ;
          sh:prefixes bamm:prefixDeclarations ;
          sh:select """
-            select $this ?properties
+            select $this ?properties ?value ?code
             where {
                $this bamm:properties ?properties .
-               ?properties rdf:rest*/rdf:first ?value .
-               filter ( isBlank( ?value ) && exists { ?value bamm:notInPayload true } && exists { ?value bamm:optional true } ) .
+               ?properties rdf:rest*/rdf:first ?property .
+               filter ( isBlank( ?property ) && exists { ?property bamm:notInPayload true } && exists { ?property bamm:optional true } ) .
+               optional {
+                 ?property bamm:property ?actualProperty .
+               }
+               bind( if( isBlank( ?property ), ?actualProperty, ?property ) as ?value )
+               bind( 'ERR_INVALID_PROPERTY' as ?code )
             }
          """
       ] ;
       sh:sparql [
          a sh:SPARQLConstraint ;
-         sh:message "A Property of an Entity may not be defined with both bamm:payloadName and bamm:notInPayload \"true\"^^xsd:boolean." ;
+         sh:message "Property '{?value}' of the Entity '{$this}' may not be defined with both bamm:payloadName and bamm:notInPayload \"true\"^^xsd:boolean." ;
```

Comment:
"Property '{?value}' of the Entity '{$this}' must not be defined as both bamm:payloadName or bamm:notInPayload \"true\"^^xsd:boolean." ;